File: ghidra_scripts/OvlReport.java
// OvlReport.java — headless report on an analyzed AC1 overlay (no PyGhidra needed)
//@category AC1
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.*;
import ghidra.app.decompiler.*;

public class OvlReport extends GhidraScript {
    public void run() throws Exception {
        FunctionManager fm = currentProgram.getFunctionManager();
        println("OVLREPORT program=" + currentProgram.getName()
                + " function_count=" + fm.getFunctionCount());
        long entryVal = currentProgram.getMemory().getInt(toAddr(0x8004ada4)) & 0xffffffffL;
        println("OVLREPORT entry_ptr=0x" + Long.toHexString(entryVal));
        Function f = fm.getFunctionContaining(toAddr(entryVal));
        println("OVLREPORT entry_function=" + (f != null ? f.getName() : "none"));
        DecompInterface di = new DecompInterface();
        di.openProgram(currentProgram);
        if (f != null) {
            DecompileResults r = di.decompileFunction(f, 60, monitor);
            if (r != null && r.getDecompiledFunction() != null) {
                println("OVLREPORT_DECOMPILE_BEGIN");
                println(r.getDecompiledFunction().getC());
                println("OVLREPORT_DECOMPILE_END");
            }
        }
    }
}

File: ghidra_scripts/ovl_report.py
# ovl_report.py — headless post-script: report analyzed overlay + decompile entry
#@category AC1
from ghidra.app.decompiler import DecompInterface
from ghidra.util.task import ConsoleTaskMonitor

fm = currentProgram.getFunctionManager()
print("=== OVL REPORT ===")
print("program:", currentProgram.getName())
print("function_count:", fm.getFunctionCount())

# entry pointer is the word at base+4
mem = currentProgram.getMemory()
af = currentProgram.getAddressFactory().getDefaultAddressSpace()
base = af.getAddress(0x8004ada0)
entry_val = mem.getInt(af.getAddress(0x8004ada4)) & 0xffffffff
print("overlay +4 entry ptr: 0x%08x" % entry_val)

ent = af.getAddress(entry_val)
fn = fm.getFunctionContaining(ent)
print("entry function:", fn.getName() if fn else "(none)")

di = DecompInterface()
di.openProgram(currentProgram)
if fn:
    res = di.decompileFunction(fn, 60, ConsoleTaskMonitor())
    if res and res.getDecompiledFunction():
        print("=== DECOMPILE entry ===")
        print(res.getDecompiledFunction().getC()[:4000])
